packages/blade/src/components/Table/_decisions/decisions.md

```diff
@@ -0,0 +1,541 @@
+
+
+#### Decision
+##### 1. Only Composable API
+- We could expose only the Composable API to the consumers 
+- This would take the least amount of implementation time and effort
+- Strong reasons to go with this approach would be the composability and tree-shaking benefits
+- Strong reasons to not go with this approach would be the verbosity and the fact that it is not the most popular approach within Table libraries ecosystem
+
+##### 2. Only Compact API
+- We could expose only the Compact API to the consumers
+- This would take the relatively more time and effort since internally we will have to build individual components and stitch them together to create the compact API
+- Internally we might not need to put in heavy efforts to ensure composable API is very well structured since we will not be exposing it to the consumers. We would use the composable API internally only to derive the final outcome of a compact API.
+- Strong reasons to go with this approach would be the popularity of the compact API within the Table libraries ecosystem
+- Strong reasons to not go with this approach would be the lack of composability and tree-shaking benefits
+
+##### 3. Both Composable & Compact API
+- We could expose both the Composable & Compact API to the consumers
+- This would take the most amount of implementation time and effort since we will have to build both the APIs
+- This will add a lot of complexity to the library and will make it harder to maintain. Any changes to the Table component, we will have to ensure both APIs are updated accordingly and always maintain parity
+- Swapping out the table library in the future would be challenging since any library we choose we will have to ensure that its feasible to build both kinds of APIs
+- Strong reasons to go with this approach would be to give our consumers complete flexibility to choose the API that they are most comfortable with
+- Strong reasons to not go with this approach would be the complexity and maintainability issues as well as education efforts required to educate the consumers about both the APIs
+
+##### Current API Usage Across Razorpay Projects
+> This is an approximate usage count. We identified this by searching through the codebase. Actual numbers may vary slightly.
+ 
+| Project         | Composable API Table Instances | Compact API Table Instances |
+| --------------- | ------------------------------ | --------------------------- |
+| PG Dashboard    | 101                            | 81                          |
+| Admin Dashboard | 19                             | 196                         |
+| X Dashboard     | 6                              | 11                          |
+
+
+##### Blade Team's Recommendation
+**We recommend** going ahead with the **Compact API** since it gives us the following advantages:
```

Comment:
We can still do `onClick` and custom rendering based on cell value with Compact API as well. There will be feature parity between both approaches. I don't foresee anything on composable that can't be done on compact and vice versa

```tsx
const columns = [{
        header: 'First Name',
        cell: (row) => <TableCell onClick={console.log}>row.firstName</TableCell>,
    }]
```

Reply:
yup yup, its possible. We had to do the same thing you've shown here.

The composable API looked more cleaner than this.

Reply:
Yeah makes sense. On the projects you work on right now can you see if developers are writing composable or compact tables?

Reply:
@anilprzp would love to hear your inputs and experience here

Reply:
Couple of more things I thought of:

- attaching refs to a table row. Since the row is internal to the table we won't be able to do that directly
- colspan attributes (I suppose this can be solved if the cell function allows only table cell)

Reply:
colspan should be easily solvable. Refs could be tricky but I'm wondering why we need refs on individual rows? Won't that be _a lot_ of refs

Reply:
Was thinking of cases where we might need to :
- reorder rows by drag and drop
- calculate the height of the row for certain animations

Reply:
Do we have any such use-cases in our products?

Reply:
I remember we needed to calculate the height of a row for some reason. But I can't remember what the use case was.